### ISSUE TITLE
Logon not working any more since 3 days

### ISSUE THREAD

**Issue by acdcnow** [DarwinsBuddy/WienerNetzeSmartmeter#45]
Integration was working will since 3 days..
Now I got the error:

Logger: homeassistant.components.sensor
Source: custom_components/wnsm/sensor.py:138
Integration: Sensor (documentation, issues)
First occurred: 05:47:49 (1 occurrences)
Last logged: 05:47:49

wnsm: Error on device update!
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 503, in _async_add_entity
    await entity.async_device_update(warning=False)
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 729, in async_device_update
    await task
  File "/config/custom_components/wnsm/sensor.py", line 195, in async_update
    zp = await self.get_zaehlpunkt(smartmeter)
  File "/config/custom_components/wnsm/sensor.py", line 138, in get_zaehlpunkt
    for z in zps[0]["zaehlpunkte"]
KeyError: 0

Redownloaded version 1.0 again

Tried to reinstall  and resetup the Integration  same issue

This error originated from a custom integration.

Logger: custom_components.wnsm.config_flow
Source: custom_components/wnsm/config_flow.py:34
Integration: WienerNetzeSmartmeter (documentation, issues)
First occurred: 06:04:27 (2 occurrences)
Last logged: 06:04:50

0
Traceback (most recent call last):
  File "/config/custom_components/wnsm/config_flow.py", line 41, in async_step_user
    zps = await self.validate_auth(
  File "/config/custom_components/wnsm/config_flow.py", line 34, in validate_auth
    return zps[0]["zaehlpunkte"] if zps is not None else []
KeyError: 0

**Comment by cleversmart** [comment]
same issue here

**Comment by DarwinsBuddy** [comment]
maybe the API changed. I'll have a look this weekend.

**Comment by reox** [comment]
Yep, they did something...
I get this error message:
```
RuntimeError: ('Cannot access zählpunkte: ', {'Exception': 'PROD API Gateway encountered an error. Error Message:  Native service provider error. Reason : com.wm.app.b2b.server.ServiceException: java.net.SocketException: Connection reset. Request Details: Service - WN_SMART_METER_PORTAL_API_B2C, Operation - /zaehlpunkte, Invocation Time:4:45:58 PM, Date:Jan 13, 2023,  Client IP - 1.2.3.4, User - Default and Application:WN_SMART_METER_PORTAL_API_B2C'})
```

I got this before rarely, but it fixed itself after a while. But this time it seems to be consistent.
Maybe they blocked all access that is not via their smartmeter portal, because on that page the requests to `/zaehlpunkte` are working :(

**Comment by ltaferner** [comment]
so maybe it's the reason why I'm not able to add the integration, because of "Username/password invalid"
tried so many things like volkszähler - not able to decrypt this data :(
hope, that @DarwinsBuddy find a solution ;)

**Comment by DarwinsBuddy** [comment]
I guess so.
When you add the integration for every Zaehlpunkt in your account a sensor will be created. each of those sensors will be initialized, hence the call to Zaehlpunkt which fail and thus you cannot add the integration.

fixing the API call's expected response will fix it.

I hope I'll find time tomorrow to have a look

**Comment by cleversmart** [comment]
Hi, if you fix the problem, maybe please add a function to retrieve historical data. In my case I miss few days and I‘m not able to add the missing data with the correct date to the recorder.

**Comment by DarwinsBuddy** [comment]
this is a home assistant feature afaik.

iirc you have to go to

Settings -> Devices/Integrations -> Entities

and at the respective sensor.you click on the most right icon to add data

Right now this may not work, as the sensor has to be up-to-date.

**Comment by DarwinsBuddy** [comment]
Found the issue.
`b2capiKey` changed to `b2cApiKey`
improved robustness of the regex looking for it

**Comment by DarwinsBuddy** [comment]
released fix in [1.0.1](https://github.com/DarwinsBuddy/WienerNetzeSmartmeter/releases/tag/v1.0.1)

**Comment by reox** [comment]
Nice! I thought already they had blocked the access :D
